test/functional/rpc_mempool_info.py

```diff
@@ -0,0 +1,103 @@
+        #      /   \       /   \
+        #    txD   txE   txF   txG
+        #            \   /
+        #             \ /
+        #             txH
+
+        def create_tx(**kwargs):
+            return self.wallet.send_self_transfer_multi(
+                from_node=self.nodes[0],
+                **kwargs,
+            )
+
+        txA = create_tx(utxos_to_spend=[confirmed_utxo], num_outputs=2)
+        txB = create_tx(utxos_to_spend=[txA["new_utxos"][0]], num_outputs=2)
+        txC = create_tx(utxos_to_spend=[txA["new_utxos"][1]], num_outputs=2)
+        txD = create_tx(utxos_to_spend=[txB["new_utxos"][0]], num_outputs=1)
+        txE = create_tx(utxos_to_spend=[txB["new_utxos"][1]], num_outputs=3)
+        txF = create_tx(utxos_to_spend=[txC["new_utxos"][0]], num_outputs=2)
+        txG = create_tx(utxos_to_spend=[txC["new_utxos"][1]], num_outputs=1)
+        txH = create_tx(utxos_to_spend=[txE["new_utxos"][2],txF["new_utxos"][0]], num_outputs=1)
+        txidA, txidB, txidC, txidD, txidE, txidF, txidG, txidH = [
+            tx["txid"] for tx in [txA, txB, txC, txD, txE, txF, txG, txH]
+        ]
+
+        mempool = self.nodes[0].getrawmempool()
+        assert_equal(len(mempool), 8)
+        for txid in [txidA, txidB, txidC, txidD, txidE, txidF, txidG, txidH]:
+            assert_equal(txid in mempool, True)
+
+        self.log.info("Find transactions spending outputs")
+        result = self.nodes[0].getmempooltxspendingprevout([ {'txid' : confirmed_utxo['txid'], 'vout' : 0}, {'txid' : txidA, 'vout' : 1} ])
+        assert_equal(result, [ {'txid' : confirmed_utxo['txid'], 'vout' : 0, 'spendingtxid' : txidA}, {'txid' : txidA, 'vout' : 1, 'spendingtxid' : txidC} ])
+
+        self.log.info("Find transaction spending multiple outputs")
+        result = self.nodes[0].getmempooltxspendingprevout([ {'txid' : txidE, 'vout' : 2}, {'txid' : txidF, 'vout' : 0} ])
+        assert_equal(result, [ {'txid' : txidE, 'vout' : 2, 'spendingtxid' : txidH}, {'txid' : txidF, 'vout' : 0, 'spendingtxid' : txidH} ])
+
+        self.log.info("Find no transaction when output is unspent")
+        result = self.nodes[0].getmempooltxspendingprevout([ {'txid' : txidH, 'vout' : 0} ])
+        assert_equal(result, [ {'txid' : txidH, 'vout' : 0} ])
+        result = self.nodes[0].getmempooltxspendingprevout([ {'txid' : txidA, 'vout' : 5} ])
+        assert_equal(result, [ {'txid' : txidA, 'vout' : 5} ])
+
+        self.log.info("Mixed spent and unspent outputs")
+        result = self.nodes[0].getmempooltxspendingprevout([ {'txid' : txidB, 'vout' : 0}, {'txid' : txidG, 'vout' : 3} ])
+        assert_equal(result, [ {'txid' : txidB, 'vout' : 0, 'spendingtxid' : txidD}, {'txid' : txidG, 'vout' : 3} ])
+
+        self.log.info("Unknown input fields")
+        result = self.nodes[0].getmempooltxspendingprevout([ {'txid' : txidC, 'vout' : 1, 'unknown' : 42} ])
+        assert_equal(result, [ {'txid' : txidC, 'vout' : 1, 'spendingtxid' : txidG} ])
```

Comment:
Question: would it be better for this to throw a `RPC_INVALID_PARAMETER`?

Reply:
Good question. I tend to silently ignore unknown parameters in APIs because it's generally more friendly to clients. If a future version of `bitcoind` introduces new optional parameters, a client can safely start including them without caring about the version of `bitcoind` it's running against, and calls will work on older `bitcoind` as well. That's not a very strong reason though, I can change that if you want.

Reply:
No objections here! Being more user-friendly sounds good :grin: